web/send_error.go

```diff
@@ -0,0 +1,57 @@
+package web
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/probs"
+)
+
+// SendError does a few things that we want for each error response:
+//  - Adds both the external and the internal error to a RequestEvent.
+//  - If the ProblemDetails provided is a ServerInternalProblem, audit logs the
+//    internal error.
+//  - Prefixes the Type field of the ProblemDetails with a namespace.
+//  - Sends an HTTP response containing the error and an error code to the user.
+func SendError(
+	log blog.Logger,
+	namespace string,
```

Comment:
Should there perhaps be a check for `namespace == ""` and a default value? Maybe overkill?

Reply:
I think this is overkill; there are just two call sites right now, and they both do it right, plus forgetting the namespace prefix isn't the worst thing that could happen (and is reasonably noticeable).